stock_putaway_product/models/product.py

```diff
@@ -0,0 +1,45 @@
+# © 2016 Jos De Graeve - Apertoso N.V. <Jos.DeGraeve@apertoso.be>
+# © 2016 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from lxml import etree
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    product_putaway_ids = fields.One2many(
+        comodel_name='stock.product.putaway.strategy',
+        inverse_name='product_tmpl_id',
+        string="Product stock locations")
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    product_putaway_ids = fields.One2many(
+        comodel_name='stock.product.putaway.strategy',
+        inverse_name='product_product_id',
+        string="Product stock locations")
+
+    @api.model
+    def fields_view_get(self, view_id=None, view_type='form',
+                        toolbar=False, submenu=False):
+        """ Custom redefinition of fields_view_get to adapt the context
+            to product variants.
+        """
+        res = super(ProductProduct, self).fields_view_get(view_id=view_id,
```

Comment:
super in python3 no longer requires explicit class and instance definition, also I'd change the formatting